File: 2022/01/py/part2.py
import sys

lines = [line.rstrip() for line in sys.stdin]

elves = [[]]
for l in lines:
    if not l:
        elves.append([])
        continue

    elves[-1].append(int(l))

count_by_elf = [sum(e) for e in elves]

print(sum(sorted(count_by_elf, reverse = True)[:3]))

File: 2022/01/ruby/part2.rb
counts = []
while line = gets
  counts << line
end
elves = counts.chunk{|l| l.chomp.empty? ? :_separator : true}

count_by_elf = elves.map do |_, elf|
  elf.map(&:to_i).sum
end

puts count_by_elf.max(3).sum

